scanpy/plotting/_tools/scatterplots.py

```diff
@@ -304,30 +356,40 @@ def embedding(
                 else:
                     in_groups_size = not_in_groups_size = size
 
-                ax.scatter(
-                    _data_points[~in_groups, 0],
-                    _data_points[~in_groups, 1],
-                    s=not_in_groups_size,
-                    marker=".",
-                    c=color_vector[~in_groups],
-                    rasterized=settings._vector_friendly,
-                    **kwargs,
+                # only show grey points if no image is below
+                if img_key is None:
+                    ax.scatter(
+                        _data_points[~in_groups, 0],
+                        _data_points[~in_groups, 1],
+                        s=not_in_groups_size,
+                        marker=".",
+                        c=color_vector[~in_groups],
+                        rasterized=settings._vector_friendly,
+                        **kwargs,
+                    )
+                scatter = (
+                    partial(ax.scatter, s=in_groups_size)
+                    if img_key is None
+                    else partial(circles, s=size_spot)
                 )
-                cax = ax.scatter(
+                cax = scatter(
                     _data_points[in_groups, 0],
                     _data_points[in_groups, 1],
-                    s=in_groups_size,
                     marker=".",
                     c=color_vector[in_groups],
                     rasterized=settings._vector_friendly,
                     **kwargs,
                 )
 
             else:
-                cax = ax.scatter(
+                scatter = (
+                    partial(ax.scatter, s=size)
+                    if img_key is None
+                    else partial(circles, s=size_spot)
+                )
```

Comment:
Why do you define it in both branches? Just define it once before the `if`.

Reply:
Right, modified